lib/handshake_crypto13.ml

```diff
@@ -7,13 +7,13 @@ let cdiv (x : int) (y : int) =
 
 let left_pad_dh group msg =
   let bytes = cdiv (Mirage_crypto_pk.Dh.modulus_size group) 8 in
-  let padding = Cstruct.create (bytes - Cstruct.length msg) in
-  padding <+> msg
+  let padding = String.make (bytes - String.length msg) '\x00' in
+  padding ^ msg
 
 let not_all_zero r =
   let* cs = r in
-  let all_zero = Cstruct.create (Cstruct.length cs) in
-  if Cstruct.equal all_zero cs then
+  let all_zero = String.make (String.length cs) '\x00' in
+  if String.equal all_zero cs then
```

Comment:
I'm sure we can compare whether everything is 0 without allocating...

Reply:
I decided to make a loop which raise an exception when we have something different than `\x00` and returns `Ok`. A double check of my implementation will be great.